executor/merge_join.go

```diff
@@ -43,169 +41,225 @@ type MergeJoinExec struct {
 	joiner       joiner
 	isOuterJoin  bool
 
-	prepared bool
-	outerIdx int
+	innerTable *mergeJoinTable
+	outerTable *mergeJoinTable
+
+	hasMatch bool
+	hasNull  bool
```

Comment:
Seems that these two fields can be removed now?

Reply:
These two fields are used to save match context when `e.joiner.tryToMatchInners` doesn't consume all the inner rows.

Reply:
You are right~